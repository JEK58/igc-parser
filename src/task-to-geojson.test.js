const readTask = require('./read-task');
const taskToGeoJSON = require('./task-to-geojson');

describe('taskToGeoJSON()', () => {
  it('converts "2017-07-15-lev" task correctly', () => {
    let task = readTask(`${__dirname}/../fixtures/2017-07-15-lev/task.tsk`);
    let geojson = taskToGeoJSON(task);
    expect(geojson).toMatchSnapshot();
  })
});
