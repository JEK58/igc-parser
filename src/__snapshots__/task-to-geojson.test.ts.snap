// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`taskToGeoJSON() converts "2017-07-15-lev" task correctly 1`] = `
Object {
  "features": Array [
    Object {
      "geometry": Object {
        "coordinates": Array [
          Array [
            6.9852833,
            51.1413833,
          ],
          Array [
            6.18695,
            50.8241667,
          ],
          Array [
            7.1375,
            50.5597167,
          ],
          Array [
            6.4916667,
            50.8888833,
          ],
          Array [
            7.2,
            51.17695,
          ],
          Array [
            7.03695,
            51.0977833,
          ],
        ],
        "type": "LineString",
      },
      "properties": Object {},
      "type": "Feature",
    },
    Object {
      "geometry": Object {
        "coordinates": Array [
          Array [
            6.9096762466963195,
            51.217654698065914,
          ],
          Array [
            7.06089035330368,
            51.06511190193409,
          ],
        ],
        "type": "LineString",
      },
      "properties": Object {},
      "type": "Feature",
    },
    Object {
      "geometry": Object {
        "coordinates": Array [
          Array [
            Array [
              6.18695,
              50.914070637726475,
            ],
            Array [
              6.200926602648388,
              50.913636892138896,
            ],
            Array [
              6.214767828540983,
              50.9123398567009,
            ],
            Array [
              6.228339634654437,
              50.91019209399071,
            ],
            Array [
              6.241510631414556,
              50.90721440408747,
            ],
            Array [
              6.2541533745632245,
              50.90343561976225,
            ],
            Array [
              6.266145615701034,
              50.89889232276094,
            ],
            Array [
              6.277371498561475,
              50.89362848414764,
            ],
            Array [
              6.287722688762933,
              50.88769503245954,
            ],
            Array [
              6.29709942562174,
              50.881149354158886,
            ],
            Array [
              6.305411485576975,
              50.874054731546266,
            ],
            Array [
              6.31257904785751,
              50.86647972391453,
            ],
            Array [
              6.318533454194282,
              50.85849749826703,
            ],
            Array [
              6.323217855625481,
              50.85018511639406,
            ],
            Array [
              6.326587740738494,
              50.841622785491815,
            ],
            Array [
              6.32861134101925,
              50.83289307981759,
            ],
            Array [
              6.329269910317621,
              50.82408014110168,
            ],
            Array [
              6.3285578767676425,
              50.81526886558006,
            ],
            Array [
              6.326482866807009,
              50.806544085574274,
            ],
            Array [
              6.3230656022060305,
              50.79798975352742,
            ],
            Array [
              6.318339672229229,
              50.789688136310424,
            ],
            Array [
              6.312351184202076,
              50.78171902744591,
            ],
            Array [
              6.305158296832491,
              50.77415898465967,
            ],
            Array [
              6.296830641635203,
              50.76708059986962,
            ],
            Array [
              6.287448638722732,
              50.76055180836163,
            ],
            Array [
              6.277102714056966,
              50.754635243487556,
            ],
            Array [
              6.265892425999464,
              50.74938764275762,
            ],
            Array [
              6.253925509657298,
              50.74485931069281,
            ],
            Array [
              6.24131684809598,
              50.74109364325793,
            ],
            Array [
              6.228187379984493,
              50.73812671811802,
            ],
            Array [
              6.214662953652414,
              50.73598695435422,
            ],
            Array [
              6.200873137878809,
              50.734694844645475,
            ],
            Array [
              6.18695,
              50.73426276227353,
            ],
            Array [
              6.173026862121192,
              50.734694844645475,
            ],
            Array [
              6.159237046347588,
              50.73598695435422,
            ],
            Array [
              6.145712620015509,
              50.73812671811802,
            ],
            Array [
              6.13258315190402,
              50.74109364325793,
            ],
            Array [
              6.119974490342703,
              50.74485931069281,
            ],
            Array [
              6.1080075740005375,
              50.74938764275762,
            ],
            Array [
              6.096797285943036,
              50.754635243487556,
            ],
            Array [
              6.08645136127727,
              50.76055180836163,
            ],
            Array [
              6.0770693583648,
              50.76708059986962,
            ],
            Array [
              6.068741703167511,
              50.77415898465967,
            ],
            Array [
              6.0615488157979245,
              50.78171902744591,
            ],
            Array [
              6.055560327770773,
              50.789688136310424,
            ],
            Array [
              6.050834397793971,
              50.79798975352742,
            ],
            Array [
              6.047417133192993,
              50.806544085574274,
            ],
            Array [
              6.045342123232359,
              50.81526886558006,
            ],
            Array [
              6.04463008968238,
              50.82408014110168,
            ],
            Array [
              6.045288658980751,
              50.83289307981759,
            ],
            Array [
              6.047312259261508,
              50.841622785491815,
            ],
            Array [
              6.0506821443745205,
              50.85018511639406,
            ],
            Array [
              6.05536654580572,
              50.85849749826703,
            ],
            Array [
              6.061320952142491,
              50.86647972391453,
            ],
            Array [
              6.068488514423025,
              50.874054731546266,
            ],
            Array [
              6.076800574378261,
              50.881149354158886,
            ],
            Array [
              6.086177311237068,
              50.88769503245954,
            ],
            Array [
              6.096528501438526,
              50.89362848414764,
            ],
            Array [
              6.107754384298968,
              50.89889232276094,
            ],
            Array [
              6.119746625436777,
              50.90343561976225,
            ],
            Array [
              6.132389368585446,
              50.90721440408747,
            ],
            Array [
              6.145560365345565,
              50.91019209399071,
            ],
            Array [
              6.159132171459018,
              50.9123398567009,
            ],
            Array [
              6.172973397351614,
              50.913636892138896,
            ],
            Array [
              6.18695,
              50.914070637726475,
            ],
          ],
        ],
        "type": "Polygon",
      },
      "properties": Object {},
      "type": "Feature",
    },
    Object {
      "geometry": Object {
        "coordinates": Array [
          Array [
            Array [
              7.1375,
              50.82942851317944,
            ],
            Array [
              7.179352818723049,
              50.82812231069425,
            ],
            Array [
              7.220795624909293,
              50.82421650072122,
            ],
            Array [
              7.261422627542617,
              50.81774934386257,
            ],
            Array [
              7.300836426986258,
              50.808784169923584,
            ],
            Array [
              7.338652082484493,
              50.79740872675282,
            ],
            Array [
              7.374501028514834,
              50.78373427978383,
            ],
            Array [
              7.408034793974462,
              50.76789447319187,
            ],
            Array [
              7.438928481745717,
              50.750043966345565,
            ],
            Array [
              7.466883970420143,
              50.730356861753684,
            ],
            Array [
              7.491632804739944,
              50.7090249429413,
            ],
            Array [
              7.512938746499308,
              50.686255742609454,
            ],
            Array [
              7.530599963090446,
              50.662270463017194,
            ],
            Array [
              7.544450836435479,
              50.63730177176441,
            ],
            Array [
              7.554363380577347,
              50.61159149704514,
            ],
            Array [
              7.560248261583971,
              50.585388246992665,
            ],
            Array [
              7.562055418538999,
              50.55894497796128,
            ],
            Array [
              7.559774289157027,
              50.53251653650726,
            ],
            Array [
              7.553433647899357,
              50.506357199467594,
            ],
            Array [
              7.543101068327476,
              50.48071823591803,
            ],
            Array [
              7.528882024785944,
              50.45584551395356,
            ],
            Array [
              7.510918651344614,
              50.431977174205876,
            ],
            Array [
              7.489388178260695,
              50.40934139082581,
            ],
            Array [
              7.464501068068155,
              50.388154239343784,
            ],
            Array [
              7.436498874802731,
              50.36861768940751,
            ],
            Array [
              7.405651850872301,
              50.3509177389073,
            ],
            Array [
              7.372256326739082,
              50.335222704459845,
            ],
            Array [
              7.336631888950181,
              50.3216816816475,
            ],
            Array [
              7.299118382196426,
              50.31042318682002,
            ],
            Array [
              7.260072761054996,
              50.301553990668054,
            ],
            Array [
              7.219865816934803,
              50.29515815218398,
            ],
            Array [
              7.178878805545934,
              50.29129626003791,
            ],
            Array [
              7.1375,
              50.29000488682058,
            ],
            Array [
              7.096121194454066,
              50.29129626003791,
            ],
            Array [
              7.0551341830651975,
              50.29515815218398,
            ],
            Array [
              7.014927238945006,
              50.301553990668054,
            ],
            Array [
              6.9758816178035765,
              50.31042318682002,
            ],
            Array [
              6.938368111049821,
              50.3216816816475,
            ],
            Array [
              6.902743673260918,
              50.335222704459845,
            ],
            Array [
              6.8693481491277,
              50.3509177389073,
            ],
            Array [
              6.83850112519727,
              50.36861768940751,
            ],
            Array [
              6.810498931931846,
              50.388154239343784,
            ],
            Array [
              6.785611821739307,
              50.40934139082581,
            ],
            Array [
              6.764081348655386,
              50.431977174205876,
            ],
            Array [
              6.746117975214056,
              50.45584551395356,
            ],
            Array [
              6.731898931672524,
              50.48071823591803,
            ],
            Array [
              6.721566352100644,
              50.506357199467594,
            ],
            Array [
              6.715225710842973,
              50.53251653650726,
            ],
            Array [
              6.712944581461002,
              50.55894497796128,
            ],
            Array [
              6.714751738416031,
              50.585388246992665,
            ],
            Array [
              6.720636619422654,
              50.61159149704514,
            ],
            Array [
              6.730549163564522,
              50.63730177176441,
            ],
            Array [
              6.744400036909556,
              50.662270463017194,
            ],
            Array [
              6.762061253500693,
              50.686255742609454,
            ],
            Array [
              6.783367195260057,
              50.7090249429413,
            ],
            Array [
              6.808116029579859,
              50.730356861753684,
            ],
            Array [
              6.836071518254285,
              50.750043966345565,
            ],
            Array [
              6.866965206025539,
              50.76789447319187,
            ],
            Array [
              6.9004989714851686,
              50.78373427978383,
            ],
            Array [
              6.9363479175155085,
              50.79740872675282,
            ],
            Array [
              6.974163573013742,
              50.808784169923584,
            ],
            Array [
              7.013577372457384,
              50.81774934386257,
            ],
            Array [
              7.054204375090706,
              50.82421650072122,
            ],
            Array [
              7.09564718127695,
              50.82812231069425,
            ],
            Array [
              7.1375,
              50.82942851317944,
            ],
          ],
        ],
        "type": "Polygon",
      },
      "properties": Object {},
      "type": "Feature",
    },
    Object {
      "geometry": Object {
        "coordinates": Array [
          Array [
            Array [
              6.4916667,
              50.978787237726486,
            ],
            Array [
              6.505662773775727,
              50.97835349021073,
            ],
            Array [
              6.5195232804037815,
              50.97705644906271,
            ],
            Array [
              6.533113988414148,
              50.97490867708133,
            ],
            Array [
              6.546303323653271,
              50.97193097470399,
            ],
            Array [
              6.558963663027207,
              50.96815217518362,
            ],
            Array [
              6.57097258685271,
              50.96360886085294,
            ],
            Array [
              6.582214076851302,
              50.95834500344499,
            ],
            Array [
              6.592579647513284,
              50.95241153222211,
            ],
            Array [
              6.601969399397452,
              50.94586583439977,
            ],
            Array [
              6.610292983901708,
              50.938771193030796,
            ],
            Array [
              6.617470470121801,
              50.931196168130164,
            ],
            Array [
              6.623433105590285,
              50.92321392736543,
            ],
            Array [
              6.628123963934883,
              50.914901533107674,
            ],
            Array [
              6.631498473794044,
              50.90633919302813,
            ],
            Array [
              6.633524824656769,
              50.897609481735294,
            ],
            Array [
              6.634184246633979,
              50.888796541173456,
            ],
            Array [
              6.63347116250117,
              50.87998526764742,
            ],
            Array [
              6.631393211659704,
              50.871260493399916,
            ],
            Array [
              6.627971146931739,
              50.86270617065085,
            ],
            Array [
              6.623238606318397,
              50.85440456591241,
            ],
            Array [
              6.617241763001535,
              50.84643547222681,
            ],
            Array [
              6.610038857947762,
              50.83887544673636,
            ],
            Array [
              6.601699620472867,
              50.83179708069494,
            ],
            Array [
              6.592304583041149,
              50.825268308669266,
            ],
            Array [
              6.581944297405023,
              50.81935176326453,
            ],
            Array [
              6.570718459934802,
              50.814104181245284,
            ],
            Array [
              6.558734954647461,
              50.80957586641631,
            ],
            Array [
              6.546108823018133,
              50.80581021408302,
            ],
            Array [
              6.532961170151526,
              50.80284330133294,
            ],
            Array [
              6.519418017305479,
              50.80070354677356,
            ],
            Array [
              6.505609111098435,
              50.79941144273203,
            ],
            Array [
              6.4916667,
              50.798979362273535,
            ],
            Array [
              6.477724288901565,
              50.79941144273203,
            ],
            Array [
              6.463915382694521,
              50.80070354677356,
            ],
            Array [
              6.450372229848473,
              50.80284330133294,
            ],
            Array [
              6.437224576981866,
              50.80581021408302,
            ],
            Array [
              6.424598445352538,
              50.80957586641631,
            ],
            Array [
              6.412614940065199,
              50.814104181245284,
            ],
            Array [
              6.401389102594977,
              50.81935176326453,
            ],
            Array [
              6.39102881695885,
              50.825268308669266,
            ],
            Array [
              6.3816337795271325,
              50.83179708069494,
            ],
            Array [
              6.373294542052238,
              50.83887544673636,
            ],
            Array [
              6.3660916369984655,
              50.84643547222681,
            ],
            Array [
              6.360094793681602,
              50.85440456591241,
            ],
            Array [
              6.355362253068261,
              50.86270617065085,
            ],
            Array [
              6.351940188340295,
              50.871260493399916,
            ],
            Array [
              6.34986223749883,
              50.87998526764742,
            ],
            Array [
              6.34914915336602,
              50.888796541173456,
            ],
            Array [
              6.349808575343231,
              50.897609481735294,
            ],
            Array [
              6.351834926205956,
              50.90633919302813,
            ],
            Array [
              6.355209436065118,
              50.914901533107674,
            ],
            Array [
              6.359900294409715,
              50.92321392736543,
            ],
            Array [
              6.365862929878198,
              50.931196168130164,
            ],
            Array [
              6.373040416098292,
              50.938771193030796,
            ],
            Array [
              6.381364000602548,
              50.94586583439977,
            ],
            Array [
              6.390753752486716,
              50.95241153222211,
            ],
            Array [
              6.4011193231486985,
              50.95834500344499,
            ],
            Array [
              6.41236081314729,
              50.96360886085294,
            ],
            Array [
              6.4243697369727935,
              50.96815217518362,
            ],
            Array [
              6.437030076346729,
              50.97193097470399,
            ],
            Array [
              6.450219411585851,
              50.97490867708133,
            ],
            Array [
              6.463810119596218,
              50.97705644906271,
            ],
            Array [
              6.477670626224273,
              50.97835349021073,
            ],
            Array [
              6.4916667,
              50.978787237726486,
            ],
          ],
        ],
        "type": "Polygon",
      },
      "properties": Object {},
      "type": "Feature",
    },
    Object {
      "geometry": Object {
        "coordinates": Array [
          Array [
            Array [
              7.2,
              51.266853937726474,
            ],
            Array [
              7.214083625004286,
              51.2664201815623,
            ],
            Array [
              7.228030826728978,
              51.2651231148033,
            ],
            Array [
              7.2417065263207725,
              51.26297530123822,
            ],
            Array [
              7.254978319636017,
              51.25999754291131,
            ],
            Array [
              7.267717779422085,
              51.25621867523699,
            ],
            Array [
              7.279801715801624,
              51.25167528318,
            ],
            Array [
              7.291113382001233,
              51.24641134147387,
            ],
            Array [
              7.301543612964758,
              51.240477782633434,
            ],
            Array [
              7.310991885338164,
              51.233931997252704,
            ],
            Array [
              7.319367288291596,
              51.226837271758384,
            ],
            Array [
              7.326589395736188,
              51.21926216940437,
            ],
            Array [
              7.332589031678677,
              51.211279860837344,
            ],
            Array [
              7.337308921714887,
              51.20296741103317,
            ],
            Array [
              7.340704224972754,
              51.19440502979369,
            ],
            Array [
              7.342742942155737,
              51.18567529330244,
            ],
            Array [
              7.3434061966880995,
              51.17686234446349,
            ],
            Array [
              7.342688387305974,
              51.168051079890084,
            ],
            Array [
              7.340597211754857,
              51.15932633147135,
            ],
            Array [
              7.337153562530165,
              51.15077205042607,
            ],
            Array [
              7.332391296819476,
              51.14247050165725,
            ],
            Array [
              7.3263568839621565,
              51.13450147605231,
            ],
            Array [
              7.319108934825814,
              51.12694152813663,
            ],
            Array [
              7.310717618502797,
              51.1198632461857,
            ],
            Array [
              7.301263972649823,
              51.11333456154081,
            ],
            Array [
              7.29083911462721,
              51.10741810345822,
            ],
            Array [
              7.2795433613405365,
              51.10217060535819,
            ],
            Array [
              7.267485266347626,
              51.09764236783339,
            ],
            Array [
              7.254780583369244,
              51.093876783231835,
            ],
            Array [
              7.241551165835624,
              51.090909926050955,
            ],
            Array [
              7.227923812515775,
              51.08877021277393,
            ],
            Array [
              7.214029069615867,
              51.08747813415014,
            ],
            Array [
              7.2,
              51.08704606227352,
            ],
            Array [
              7.185970930384134,
              51.08747813415014,
            ],
            Array [
              7.172076187484225,
              51.08877021277393,
            ],
            Array [
              7.158448834164377,
              51.090909926050955,
            ],
            Array [
              7.145219416630757,
              51.093876783231835,
            ],
            Array [
              7.132514733652375,
              51.09764236783339,
            ],
            Array [
              7.120456638659465,
              51.10217060535819,
            ],
            Array [
              7.109160885372791,
              51.10741810345822,
            ],
            Array [
              7.098736027350178,
              51.11333456154081,
            ],
            Array [
              7.089282381497204,
              51.1198632461857,
            ],
            Array [
              7.080891065174187,
              51.12694152813663,
            ],
            Array [
              7.073643116037845,
              51.13450147605231,
            ],
            Array [
              7.067608703180525,
              51.14247050165725,
            ],
            Array [
              7.062846437469835,
              51.15077205042607,
            ],
            Array [
              7.059402788245145,
              51.15932633147135,
            ],
            Array [
              7.057311612694027,
              51.168051079890084,
            ],
            Array [
              7.056593803311902,
              51.17686234446349,
            ],
            Array [
              7.057257057844264,
              51.18567529330244,
            ],
            Array [
              7.0592957750272465,
              51.19440502979369,
            ],
            Array [
              7.062691078285114,
              51.20296741103317,
            ],
            Array [
              7.067410968321324,
              51.211279860837344,
            ],
            Array [
              7.073410604263813,
              51.21926216940437,
            ],
            Array [
              7.080632711708405,
              51.226837271758384,
            ],
            Array [
              7.089008114661836,
              51.233931997252704,
            ],
            Array [
              7.0984563870352435,
              51.240477782633434,
            ],
            Array [
              7.1088866179987695,
              51.24641134147387,
            ],
            Array [
              7.120198284198378,
              51.25167528318,
            ],
            Array [
              7.132282220577916,
              51.25621867523699,
            ],
            Array [
              7.1450216803639846,
              51.25999754291131,
            ],
            Array [
              7.15829347367923,
              51.26297530123822,
            ],
            Array [
              7.171969173271023,
              51.2651231148033,
            ],
            Array [
              7.185916374995715,
              51.2664201815623,
            ],
            Array [
              7.2,
              51.266853937726474,
            ],
          ],
        ],
        "type": "Polygon",
      },
      "properties": Object {},
      "type": "Feature",
    },
    Object {
      "geometry": Object {
        "coordinates": Array [
          Array [
            7.015140937337296,
            51.11557074041413,
          ],
          Array [
            7.058759062662704,
            51.07999585958588,
          ],
        ],
        "type": "LineString",
      },
      "properties": Object {},
      "type": "Feature",
    },
  ],
  "type": "FeatureCollection",
}
`;
